### PR TITLE
Refactor Effects + changes

### PR DESCRIPTION
**Pull request by jfboeve**
- Refactored the Effects to the ShaderManager, this way a developer can add their own custom effects.
- Added more docs on the FadeOutEffect and RadiusEffect.
- *breaking* Angle property is now in Radians not in Degrees. The reason for this change is that we only use radians in the core, and it saves a calculation for each pixel from degrees to radians. The dynamic-shader example has a example function that calculates degrees to radians (PI / 180 * degrees)

### CODE REVIEW

File: src/core/renderers/webgl/shaders/effects/RadiusEffect.ts [hunk 1]
```diff
@@ -29,6 +29,20 @@ export interface RadiusEffectProps extends DefaultEffectProps {
   /**
    * Corner radius, in pixels, to cut out of the corners
    *
+   * You can input an array with a length of up to four or a number.
```
Comment (frank-weindel) [inline src/core/renderers/webgl/shaders/effects/RadiusEffect.ts:32]: Just for the future, it's better to place all these longer pieces of explanation under a `@remarks` header.

File: src/core/CoreShaderManager.ts [hunk 2]
```diff
@@ -62,6 +103,13 @@ export class CoreShaderManager {
     this.shConstructors[shType] = shClass;
   }
 
+  registerEffectType<Type extends keyof EffectMap>(
```
Comment (frank-weindel) [inline src/core/CoreShaderManager.ts:106]: Is the `ShaderEffect` base class exported via the Core API export (exports/core-api.ts)? I don't think it is at first glance. Users will need that as well in order to create custom effects.

File: src/core/renderers/webgl/shaders/effects/RadiusEffect.ts (hunk 1 above)
Comment (jfboeve) [reply src/core/renderers/webgl/shaders/effects/RadiusEffect.ts:33]: done

File: src/core/CoreShaderManager.ts (hunk 2 above)
Comment (jfboeve) [reply src/core/CoreShaderManager.ts:106]: Done

**Pull request merged**